### DISCUSSION TITLE
Replacing F5 BigIP functionality with YARP

### DISCUSSION THREAD

**Discussion by mrob1600** [dotnet/yarp#362] [General] [answered]
Hoping to use YARP for HA/load balancing to replace BigIP for our in-house built applications. I'm not clear on how to specify destination addresses with YARP.  With BigIP we specify IP address and port for the destination servers and the request flows through as-is (simple scenarios) to one of the servers. 

So we have two identical IIS servers with an application binding of myapplication.com (port 443). We want the user to enter myapplication.com in the browser to access the application. https://myapplication.com resolves to a virtual IP address on BigIP and then BigIP sends the request (using SSL passthrough) to one of the destination servers based on the specified load balancing algorithm. IIS selects the correct web site based on the incoming host name of myapplication.com. 

What would be the destination addresses for the two servers in this example? We need the actual URL of the request to flow through to the backend servers.

**Comment by Tratcher** [comment, marked as answer]
Have you looked at the getting started guide?
https://microsoft.github.io/reverse-proxy/articles/getting_started.html

Destinations are specified in this section, you specify the scheme, host/IP, and port as a URI:
```
     "cluster2": {
       "Destinations": {
         "cluster2_destination1": {
           "Address": "https://192.168.0.1:45324/"
         }
       }
```

> We need the actual URL of the request to flow through to the backend servers.

Note the full URL isn't included in one place in the HTTP protocol, it's broken down into parts. The path and query will flow by default. You specify the scheme to use in the destination address. The host becomes the Host header. YARP uses the destination host in the Host header by default, but you can tell it to use the original request host with the following transform:
https://microsoft.github.io/reverse-proxy/articles/transforms.html#requestheaderoriginalhost
```
{
  "ReverseProxy": {
    "Routes": [
      {
        "RouteId": "route1",
        "ClusterId": "cluster1",
        "Match": {
          "Hosts": [ "myapplication.com" ]
        },
        "Transforms": [
          { "RequestHeaderOriginalHost": "true" },
...
```

**Reply by mrob1600** [reply]
The transform to use the original host header is what I was looking for. Just didn't look hard enough. Thanks for your help!
